### PR TITLE
Add checks for zookeeper 3.5

### PR DESCRIPTION
**Pull request by duncaan**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out on [Our CHANGELOG Guidelines ](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

- [x] Tests

- [x] Add the plugin to the README

- [x] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose
Use the new netty endpoint for metrics instead of the sockets which have to be explicitly enabled in zookeeper 3.5.

### CODE REVIEW

**Comment by majormoses** [comment]
What version does the current check support? Would it be possible to combine the two checks and maybe auto detect version and do things differently based on the version? If not what I'd like to see first is some effort to split out more logic from those checks as I see quite a bit of duplicated code.

**Comment by duncaan** [comment]
I tested this with zookeeper 3.5.3 (latest beta).  To me, this doesn't have anything to do with the other cluster check since it does not rely on a curator instance.  The checks aren't really the same, this check queries the new netty endpoint that is enabled by default, not the socket as used by other checks.

**Comment by majormoses** [comment]
OK I have not run zookeeper so I am not sure I am understanding this correctly so apologies if I got this wrong. The existing checks rely on a service called `curator` that makes using `zookeeper` easier to use but are not required. In an ideal world we would rename all the existing checks to be something like `check-curator-some-name.rb` so designate that they require this and yours would drop the curator prefix. Unfortunately I could see that being painful for existing users to change. I think this leaves us with a few options:
- do a minor release deprecating all the existing checks by renaming them and replacing them with a `binstub` that prints a deprecation warning and then executes the renamed check. After this we can do a major version bump removing the binstubs, freeing up up the natural namespace. After that you would need to rebase your branch, and remove the `3.5` from the name.
- use a prefix of `netty-` or something else that makes more sense to differentiate them.

**Comment by duncaan** [comment]
I'm more than happy with changing the name of my check so that it's useful for those that upgrade to zk 3.5.  I think that's the path of least resistance, at least.

**Comment by majormoses** [comment]
sorry can we do `check-netty-` sorry I was not clear enough. We always start the scripts with `check-`, `metric-`, `mutator-`, `handler-`, etc to make it easy to script against and make it obvious to its intention.

File: bin/check-netty-zookeeper-cluster.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  option :latency,
+         description: 'Critical threshold for Zookeeper average latency',
+         short: '-l TICKS',
+         long: '--latency TICKS',
+         proc: proc(&:to_i),
+         default: 10
+
+  def get_monitoring_output(port)
+    r = RestClient::Resource.new("http://localhost:#{port}/commands/monitor",
```
Comment (majormoses) [inline bin/check-netty-zookeeper-cluster.rb]: I don't see the `rest-client` gem in the `.gemspec` we would need to add it there to ensure that the dependencies will be met at runtime. Anyways I'd prefer to use `Net::HTTP` as `rest-client` would add the dependency of a `c` compiler.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-zookeeper/versions/1.5.0

**Pull request merged**